docs/manifest-devhub.json

```diff
@@ -653,6 +653,12 @@
 		"markdown_source": "../packages/components/src/color-picker/README.md",
 		"parent": "components"
 	},
+	{
+		"title": "ComboboxControl",
```

Comment:
Cool stuff in this PR 😍 

It's mostly for @youknowriad, the keeper of the blocks (and sometimes components). It would be nice to consolidate all similar listed controls into one API. We have at the moment:
- `SelectControl` - very basic one, a wrapper for `select` tag
- `CustomSelectControl` - more advanced version of `select` with custom options
- `ComboboxControl` - this PR, searchable version
- `Autocomplete` - in many ways similar to the control from this PR but very specific to RichText

Reply:
I think these are fundamentally different components. `select !== combobox`. Converging them would only lead to a bloated interface and a less clear separation of concerns and design guidelines.

The only thing I think could be consolidated here is to use `ComboboxControl` in `Autocomplete` somehow.

Reply:
Interesting thoughts @gziolo + @epiqueras !

On the surface, it may appear like these very similar'ish components should be consolidated. As @epiqueras pointed out, some of them are different things.

---

Fundamentally, I think there are 2 select-based primitives:
`SelectControl` and `CustomSelectControl`.

One uses native HTML `select`, the other provides a custom ID.

The native HTML `select` functionality is limited to what the browser provides.
The custom one can be enhanced/extended as we need.

Foundationally, the custom one should support all the functionality of the native HTML `select`, with enhancements like custom rendering, searching, filtering.

With the implementation of [downshift](https://github.com/downshift-js/downshift), it does all these things (❤️ )

A [ComboBox](https://en.wikipedia.org/wiki/Combo_box) is a well recognized UI element/pattern, independent of [Select](https://en.wikipedia.org/wiki/Drop-down_list).

Because we have the `CustomSelectControl`, `ComboBox` can be constructed using that as the foundation.

Likewise, [Autocomplete](https://en.wikipedia.org/wiki/Autocomplete) can be a different mechanic. Which, like `ComboBox`, can be constructed with `CustomSelectControl`.

---

In the end, we'll have several UI elements to accommodate different common use-cases. However, all of them would be constructed from 2 selection primitives.

That's how I see it 😊

Reply:
The interesting part here is that Downshift is advertised as a tool to cover it all:

> Primitives to build simple, flexible, WAI-ARIA compliant React autocomplete/dropdown/select/combobox components

which goes in line with what @ItsJonQ said:

> In the end, we'll have several UI elements to accommodate different common use-cases. However, all of them would be constructed from 2 selection primitives.

It's inevitable that we are going to share the same logic in those components as described.

My only concern is the maintenance cost of 4 distinct public APIs introduced in this space. It's always the challenge to keep the docs up to date, make those components easy to discover for developers who want to pick the best tool to achieve their goal. If you feel like this is the best way to go than I'm happy with the proposal. Thanks for taking the time to respond to my concern.

Again, I like this new component a lot 👍

Reply:
Downshift is a single component because it's a single state manager with a shared interface.

Component implementations that use it, usually are not single components, because of markup and other non-core logic not being easy to share.

Reply:
I agree that `select` and `combobox` are fundamentally different:

* When clicked, `select` lets us choose from a fixed list of options, while `combobox` shows only a few of the available options after text is entered in the input;
* Because of that difference the custom `select` uses a button while the `combobox` uses an input;
* Downshift recognises this by making two different hooks available, for each use case.

In addition to this, I'd be concerned that shoehorning these into one component would make the task of ensuring full accessibility harder than it already is for each individual one, as they each have different requirements.